src/isomorphic/deprecated/__tests__/cloneWithProps-test.js

```diff
@@ -30,6 +30,26 @@ describe('cloneWithProps', function() {
     onlyChild = require('onlyChild');
     cloneWithProps = require('cloneWithProps');
     emptyObject = require('emptyObject');
+    spyOn(console, 'error');
+  });
+
+  it('should warn because it is deprecated', function() {
+    var Grandparent = React.createClass({
+      render: function() {
+        return <Parent><div /></Parent>;
+      },
+    });
+    var Parent = React.createClass({
+      render: function() {
+        return (
+          <div>
+            {cloneWithProps(onlyChild(this.props.children), {})}
+          </div>
+        );
+      },
+    });
+    var component = ReactTestUtils.renderIntoDocument(<Grandparent />);
+    expect(console.error.argsForCall.length).toBe(1);
```

Comment:
We should also assert the text of the error message (ie. `cloneWithProps(...) is deprecated. Please use React.cloneElement instead.`) because that makes it easier to debug when the test starts failing.